website/docs/mutation.md

```diff
@@ -213,6 +214,43 @@ spec:
 - `spec.parameters.values.fromList` holds the list of values that will be added or removed.
 - `operation` can be `merge` to insert values into the list if missing, or `prune` to remove values from the list. `merge` is default.
 
+### AssignImage
+
+AssignImage is a mutator specifically for changing the components of an image
+string. Suppose you have an image like `my.registry.io:2000/repo/app:latest`.
+`my.registry.io:2000` would be the domain, `repo/app` would be the path, and
+`:latest` would be the tag. The domain, path, and tag of an image can be changed
+separately or in conjunction.
+
+For example, to change the whole image to `my.registry.io/repo/app@sha256:abcde67890123456789abc345678901a`:
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: AssignImage
+metadata:
+  name: assign-container-image
+spec:
+  applyTo:
+  - groups: [ "" ]
+    kinds: [ "Pod" ]
+    versions: [ "v1" ]
+  location: "spec.containers[name:*].image"
+  parameters:
+    assignDomain: "my.registry.io"
+    assignPath: "repo/app"
+    assignTag: "@sha256:abcde67890123456789abc345678901a"
+  match:
+    source: "All"
```

Comment:
nit: would be good to also include an example that matched `Deployment` with expansion

Reply:
What do you mean by "matched `Deployment` with expansion?

If one were to use `AssignImage` in a context where it only is applied during expansion, one would have to set `source: Generated` and then target the expanded resources (i.e Pod), not the Deployment. This example has `source: "All"`, so it would apply to expanded Pods as well.

Reply:
I saw you LGTM, so I'm going to merge. If a doc change comes to mind, let me know and we can do it in a follow up PR :)